### ISSUE TITLE
Not working on Windows again

### ISSUE THREAD

**Issue by ribbles** [Voxer/sse4_crc32#46]
```
  crc32c_sse42.cpp
..\src\crc32c_sse42.cpp(52): error C3861: '_mm_crc32_u8': identifier not found [C:\git\main\node_modules\sse4_crc32\build\crc32c_sse42.vcxproj]
..\src\crc32c_sse42.cpp(57): error C3861: '_mm_crc32_u64': identifier not found [C:\git\main\node_modules\sse4_crc32\build\crc32c_sse42.vcxproj]
..\src\crc32c_sse42.cpp(59): error C3861: '_mm_crc32_u32': identifier not found [C:\git\main\node_modules\sse4_crc32\build\crc32c_sse42.vcxproj]
..\src\crc32c_sse42.cpp(60): error C3861: '_mm_crc32_u16': identifier not found [C:\git\main\node_modules\sse4_crc32\build\crc32c_sse42.vcxproj]
..\src\crc32c_sse42.cpp(61): error C3861: '_mm_crc32_u8': identifier not found [C:\git\main\node_modules\sse4_crc32\build\crc32c_sse42.vcxproj]
```

**Comment by anandsuresh** [comment]
@ribbles Can you send more info about the build environment? Windows/VC++/Studio/node/npm versions, etc.

**Comment by akashkrishnan** [comment]
I've been getting the same issue for several weeks now.

OS: Windows 10
VC++: 2013 and 2015
Node: 4.1.1
npm: 2.14.6

**Comment by anandsuresh** [comment]
@ribbles @akashkrishnan What module are you trying to install? Is it the `node-gcloud` module? Also, I'd need logs for your npm install process to figure out more stuff.

**Comment by akashkrishnan** [comment]
Yes, it is the `gcloud` module. Where would I find the npm logs? A debug log isn't created by npm in the root directory.

Running `npm install --save gcloud` results in the following console output:

```
> sse4_crc32@4.1.0 install C:\simpleemotion\sedemo.tk\node_modules\gcloud\node_modules\hash-stream-validation\node_modules\fast-crc32c\node_modules\sse4_crc32
> node-gyp rebuild


C:\simpleemotion\sedemo.tk\node_modules\gcloud\node_modules\hash-stream-validation\node_modules\fast-crc32c\node_modules\sse4_crc32>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  crc32c_sse42.cpp
..\src\crc32c_sse42.cpp(52): error C3861: '_mm_crc32_u8': identifier not found [C:\simpleemotion\sedemo.tk\node_modules\gcloud\node_modules\hash-stream-validation\node_modules\fast-crc32c\node_modules\sse4_crc32\build\crc32c_sse42.vcxproj]
..\src\crc32c_sse42.cpp(57): error C3861: '_mm_crc32_u64': identifier not found [C:\simpleemotion\sedemo.tk\node_modules\gcloud\node_modules\hash-stream-validation\node_modules\fast-crc32c\node_modules\sse4_crc32\build\crc32c_sse42.vcxproj]
..\src\crc32c_sse42.cpp(59): error C3861: '_mm_crc32_u32': identifier not found [C:\simpleemotion\sedemo.tk\node_modules\gcloud\node_modules\hash-stream-validation\node_modules\fast-crc32c\node_modules\sse4_crc32\build\crc32c_sse42.vcxproj]
..\src\crc32c_sse42.cpp(60): error C3861: '_mm_crc32_u16': identifier not found [C:\simpleemotion\sedemo.tk\node_modules\gcloud\node_modules\hash-stream-validation\node_modules\fast-crc32c\node_modules\sse4_crc32\build\crc32c_sse42.vcxproj]
..\src\crc32c_sse42.cpp(61): error C3861: '_mm_crc32_u8': identifier not found [C:\simpleemotion\sedemo.tk\node_modules\gcloud\node_modules\hash-stream-validation\node_modules\fast-crc32c\node_modules\sse4_crc32\build\crc32c_sse42.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\12.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 10.0.10240
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\simpleemotion\sedemo.tk\node_modules\gcloud\node_modules\hash-stream-validation\node_modules\fast-crc32c\node_modules\sse4_crc32
gyp ERR! node -v v4.1.1
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok
npm WARN optional dep failed, continuing sse4_crc32@4.1.0
gcloud@0.23.0 node_modules\gcloud
├── propprop@0.3.0
├── string-format-obj@1.0.1
├── methmeth@1.1.0
├── prop-assign@1.0.0
├── arrify@1.0.0
├── async@1.4.2
├── is@3.1.0
├── extend@3.0.0
├── dns-zonefile@0.1.10
├── create-error-class@2.0.1 (capture-stack-trace@1.0.0, inherits@2.0.1)
├── stream-events@1.0.1 (stubs@1.1.2)
├── stream-forward@3.0.0 (stubs@2.0.0)
├── once@1.3.2 (wrappy@1.0.1)
├── mime-types@2.1.7 (mime-db@1.19.0)
├── split-array-stream@1.0.0 (is-stream-ended@0.1.0)
├── retry-request@1.2.1 (stream-cache@0.0.2)
├── pumpify@1.3.3 (pump@1.0.0)
├── through2@2.0.0 (xtend@4.0.0, readable-stream@2.0.2)
├── duplexify@3.4.2 (end-of-stream@1.0.0, readable-stream@2.0.2)
├── concat-stream@1.5.0 (inherits@2.0.1, typedarray@0.0.6, readable-stream@2.0.2)
├── gcs-resumable-upload@0.2.2 (buffer-equal@0.0.1, configstore@1.2.1)
├── request@2.64.0 (aws-sign2@0.5.0, forever-agent@0.6.1, tunnel-agent@0.4.1, oauth-sign@0.8.0, form-data@1.0.0-rc3, caseless@0.11.0, stringstream@0.0.4, isstream@0.1.2, json-stringify-safe@5.0.1, tough-cookie@2.1.0, qs@5.1.0, node-uuid@1.4.3, combined-stream@1.0.5, http-signature@0.11.0, bl@1.0.0, hawk@3.1.0, har-validator@1.8.0)
├── hash-stream-validation@0.1.2 (fast-crc32c@1.0.0)
├── google-auto-auth@0.2.2 (object-assign@3.0.0, google-auth-library@0.9.6)
├── gce-images@0.2.0 (object-assign@3.0.0, got@4.2.0, google-auto-auth@0.1.1)
└── protobufjs@4.1.1 (ascli@1.0.0, bytebuffer@4.1.0, glob@5.0.15, yargs@3.26.0)
```

**Comment by anandsuresh** [comment]
@rvagg @kkoopa Does Nan v2 support older versions of Node.js, mainly 0.8, 0.10 and 0.12? I have a feeling that most of these issues that are being reported are a result of version mismatches (i.e. people using an incorrect version of the sse4_crc32 library).

**Comment by anandsuresh** [comment]
@ribbles @akashkrishnan Please standby. I think I'll need to push some changes to the gcloud module to get this to work.

Also, I don't have access to a Windows 10 machine. The best I've been able to test on the Windows side was Windows 8.

**Comment by kkoopa** [comment]
Yes

**Comment by anandsuresh** [comment]
Thanks @kkoopa!

@akashkrishnan What version of gcloud are you trying to build? Can you please use the latest version?

**Comment by akashkrishnan** [comment]
@anandsuresh I get the same issue with `npm install --save gcloud@latest`. The `package.json` under my gcloud installation also says `0.23.0`, which is the latest.

**Comment by Ryanc1256** [comment]
@anandsuresh I am having trouble as well, windows 10... I get this error...


..\src\crc32c_sse42.cpp(52): error C3861: '_mm_crc32_u8': identifier not found [node_modules
\sse4_crc32\build\crc32c_sse42.vcxproj]
..\src\crc32c_sse42.cpp(57): error C3861: '_mm_crc32_u64': identifier not found [\node_module
s\sse4_crc32\build\crc32c_sse42.vcxproj]
..\src\crc32c_sse42.cpp(59): error C3861: '_mm_crc32_u32': identifier not found [\node_module
s\sse4_crc32\build\crc32c_sse42.vcxproj]
..\src\crc32c_sse42.cpp(60): error C3861: '_mm_crc32_u16': identifier not found [\node_module
s\sse4_crc32\build\crc32c_sse42.vcxproj]
..\src\crc32c_sse42.cpp(61): error C3861: '_mm_crc32_u8': identifier not found [\node_modules
\sse4_crc32\build\crc32c_sse42.vcxproj]

**Comment by anandsuresh** [comment]
@Ryanc1256 Versions in use? NPM, Node, MsBuild, VC++, Windows etc.

**Comment by anandsuresh** [comment]
Also, can you paste logs for the `npm install` process as well?

**Comment by Ryanc1256** [comment]
node: 4.2.1
npm: 3.3.10
Visual Studio 2015
Windows 10

> sse4_crc32@4.1.0 install C:\Users\the-d\Desktop\New folder\test\node_modules\sse4_crc32
> node-gyp rebuild

```
C:\Users\the-d\Desktop\New folder\test\node_modules\sse4_crc32>if not defined npm_config_node_gyp (node "C:\Users\the-d\AppData\Roaming\npm\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  crc32c_sse42.cpp
..\src\crc32c_sse42.cpp(52): error C3861: '_mm_crc32_u8': identifier not found [C:\Users\the-d\Desktop\New folder\test\node_modules\sse4_crc32\build\crc32c_sse42.vcxproj]
..\src\crc32c_sse42.cpp(57): error C3861: '_mm_crc32_u64': identifier not found [C:\Users\the-d\Desktop\New folder\test\node_modules\sse4_crc32\build\crc32c_sse42.vcxproj]
..\src\crc32c_sse42.cpp(59): error C3861: '_mm_crc32_u32': identifier not found [C:\Users\the-d\Desktop\New folder\test\node_modules\sse4_crc32\build\crc32c_sse42.vcxproj]
..\src\crc32c_sse42.cpp(60): error C3861: '_mm_crc32_u16': identifier not found [C:\Users\the-d\Desktop\New folder\test\node_modules\sse4_crc32\build\crc32c_sse42.vcxproj]
..\src\crc32c_sse42.cpp(61): error C3861: '_mm_crc32_u8': identifier not found [C:\Users\the-d\Desktop\New folder\test\node_modules\sse4_crc32\build\crc32c_sse42.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\the-d\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 10.0.10240
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\the-d\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\the-d\Desktop\New folder\test\node_modules\sse4_crc32
gyp ERR! node -v v4.2.1
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok
npm WARN install:sse4_crc32@4.1.0 sse4_crc32@4.1.0 install: `node-gyp rebuild`
npm WARN install:sse4_crc32@4.1.0 Exit status 1
```

**Comment by kkoopa** [comment]
You're including the wrong header. `CRC32` is part of SSE 4.2 which lies in `nmmintrin.h`. `smmintrin.h` is for SSE 4.1.

**Comment by anandsuresh** [comment]
Whoops!!! I wonder how long it has been that way!! Probably since the very first version when I was testing it out on SmartOS. Surprisingly it works fine on SmartOS, FreeBSD, and Linux with `smmintrin.h`. Thanks for that catch @kkoopa.

@Ryanc1256 @ribbles @akashkrishnan Please clone this repo, and try running `npm install` on the `anand-intrinsics` branch to see if that works.

**Comment by Ryanc1256** [comment]
@anandsuresh yeah that worked... looks good

**Comment by anandsuresh** [comment]
Sweet! Closing this out and release a new version with the fixes.

**Comment by anandsuresh** [comment]
Released sse4_crc32 v4.1.1.

**Comment by JunielKatarn** [comment]
@kkoopa Thanks fot that. That include solves issues in Voxer-unrelated projects too :)
